src/irmin-pack/unix/file_manager.ml

```diff
@@ -566,4 +567,28 @@ struct
     let* () = Suffix.close old_suffix in
     let* () = if unlink then Io.unlink (Suffix.path old_suffix) else Ok () in
     Ok ()
+
+  let write_gc_output ~root ~generation output =
+    let open Result_syntax in
+    let path = Irmin_pack.Layout.V3.gc_result ~root ~generation in
+    let* io = Io.create ~path ~overwrite:true in
+    let out = Errs.to_json_string output in
+    let* () = Io.write_string io ~off:Int63.zero out in
+    Io.close io
+
+  let read_gc_output ~unlink ~root ~generation =
+    let open Result_syntax in
+    let read_file () =
+      let path = Irmin_pack.Layout.V3.gc_result ~root ~generation in
+      let* io = Io.open_ ~path ~readonly:true in
+      let* len = Io.read_size io in
+      let len = Int63.to_int len in
+      let* string = Io.read_to_string io ~off:Int63.zero ~len in
+      let* () = Io.close io in
+      let* () = if unlink then Io.unlink path else Ok () in
+      Ok string
+    in
+    match read_file () with
+    | Ok x -> (Ok (), Errs.of_json_string x)
+    | Error err -> (Error err, Ok 0)
```

Comment:
Would it be possible to merge the pair of results into a single result? I feel such a function either fully succeeds (Ok) or not (Error).